backend/app/service/chat_service.py

```diff
@@ -124,12 +126,112 @@ async def step_solve(options: Chat, request: Request, task_lock: TaskLock):
+                    import time
+                    # Generate unique task ID 
+                    task_id = item.data.get('task_id', f"{int(time.time() * 1000)}-multi")
+                    # Create and assign new task immediately
+                    new_camel_task = Task(content=new_task_content, id=task_id)
+                    # Preserve additional_info from the previous task if it exists
+                    if hasattr(camel_task, 'additional_info') and camel_task.additional_info:
+                        new_camel_task.additional_info = camel_task.additional_info
+                    camel_task = new_camel_task
+                
+                # Now trigger end of previous task using stored result
+                yield sse_json("end", old_task_result)
+                # Always yield new_task_state first - this is not optional
+                yield sse_json("new_task_state", item.data)
+                # Trigger Queue Removal
+                yield sse_json("remove_task", {"task_id": item.data.get("task_id")})
+                
+                # Then handle multi-turn processing
+                if workforce is not None and new_task_content:
+                    logger.info(f"[CHAT] MULTI-TURN: Processing new task {item.data.get('task_id')}")
+                    task_lock.status = Status.confirming
+                    workforce.pause()
+                    
+                    try:
+                        yield sse_json("confirmed", "")
+                        task_lock.status = Status.confirmed
+                        
+                        # Use existing workforce to decompose (without creating new one)
+                        # Append to _pending_tasks
+                        new_sub_tasks = await workforce.handle_decompose_append_task(
+                            camel_task,  # Use the updated camel_task
+                            reset=False  # Keep existing agents and context
+                        )
+                        
+                        # Generate summary using existing agents
+                        summary_task_agent_instance = task_summary_agent(options)
+                        new_summary_content = await summary_task(summary_task_agent_instance, camel_task)
+                        
+                        # Send the extracted events
+                        yield to_sub_tasks(camel_task, new_summary_content)
+                        
+                        # Update the context with new task data
+                        sub_tasks = new_sub_tasks
+                        summary_task_content = new_summary_content
+                        
+                        logger.info(f"[CHAT] Multi-turn task decomposed into {len(sub_tasks)} subtasks")
+                        
+                    except Exception as e:
+                        logger.error(f"[CHAT] Error processing multi-turn task: {e}")
+                        # Continue with existing context if decomposition fails
```

Comment:
we can notify user of the error 
```suggestion
                        
                    except Exception as e:
                        logger.error(f"[CHAT] Error processing multi-turn task: {e}")
                        # Continue with existing context if decomposition fails
                        yield sse_json("error", {"message": f"Failed to process task: {str(e)}"})
```

Reply:
Good suggestion